### DISCUSSION TITLE
Folders for modules

### DISCUSSION THREAD

**Discussion by erick-prosimo** [terrakube-io/terrakube#303] [Ideas]
It seems today a module is 1:1 with a repo, but it would be great to support directories. The way it stands today (as far as I could tell) it requires many repos all with a specific name format to support multiple modules.

**Comment by alfespa17** [comment]
This is a desing choice, whey you have a private terraform registry like in Terrakube you can have one repository for the module and you handle different versions of the module using different "releases/tags" and that helps when you are using a lot of workspaces because if you do a change in your module your workspaces could still use old version and wont be affected by new changes. In the long term it is easier to maintain. 

We support folders when you use the workspaces not in modules

**Comment by alfespa17** [comment]
Folder support for modules was added in this issue and the feature will be available in version 2.20.0

https://github.com/AzBuilder/terrakube/issues/653
